src/nvim/eval.c

```diff
@@ -6531,13 +6531,18 @@ static void f_abs(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 static void f_add(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 {
   list_T      *l;
+  const char *const arg_errmsg = N_("add() argument");
```

Comment:
You don’t need the variable unless it is used in multiple places.

Reply:
I see it actually could be used in multiple places. Though I think it is better to remove `else` clause, as well as the check for NULL and add `tv_list_locked` function near `tv_list_len` with similar contents and purpose: in any case I was planning to eventually remove any references outside of typval.[ch] to any list and dictionary internal structure members so it would be possible to play with their implementation.

Reply:
I have two questions regarding this comment:
First about removing the else clause: I'm not sure to which occurrence of `tv_list_len` you refer. My interpretation currently is that instead of the current if {if something/else error} else error, you would like the errors in f_add to be grouped right? So more like if () {error} else {perform actions}. Is this what you mean?
Second about 'misusing' `tv_check_fixed` to generate an error using `VAR_FIXED`. I could duplicate the define of 'E742' from the `tv_check_fixed` to the top of eval.c, where `e_listreq` is also defined (and perhaps sort those on error number), and then replace all those misuses, reducing the amount of calls to tv_ functions by 3. Do you like the idea?

Reply:
You added only one else to this function, it can be removed if you replace `l->lv_lock` with (to be created) `tv_list_locked` and remove the NULL check above. Do not duplicate the define, use `tv_check_lock`, but only once.